packages/ts/models/src/ConstraintBuilder.ts

```diff
@@ -0,0 +1,111 @@
+import type { EmptyObject } from 'type-fest';
+import {
+  $assertSupportedModel,
+  type Constraint,
+  type ConstraintFn,
+  type NonAttributedConstraint,
+  $defaultValue,
+  $name,
+  type AnyObject,
+  Model,
+  type Value,
+} from './Model.js';
+
+/**
+ * A builder class for declaring constraints.
+ * @internal low-level API for internal use only.
+ *
+ * @typeParam V - The value type of the constraint.
+ * @typeParam N - The name of the constraint.
+ * @typeParam A - The attributes for the constraint function. It is an object with string
```

Comment:
Sorry, I didn't get the description. I guess it should be made a bit clearer.

So, what is the attribute object, how it is connected to the constraint function, how they are interconnected?